### ISSUE TITLE
[bug] SIWE Broken in version 4.1.0

### ISSUE THREAD

**Issue by joeblau** [WalletConnect/web3modal#2044]
### Link to minimal reproducible example

https://github.com/joeblau/web3modal-4.1-bug

### Summary

Documentation Issues

https://docs.walletconnect.com/web3modal/nextjs/siwe/next-auth

```diff
- import type { SIWESession } from '@web3modal/core'
+ import type { SIWESession } from '@web3modal/core'
```

```diff
- import type { SIWECreateMessageArgs, SIWESession, SIWEVerifyMessageArgs } from "@web3modal/core";
+ import type { SIWECreateMessageArgs, SIWESession, SIWEVerifyMessageArgs } from "@web3modal/siwe";
```

The sample project that implements SIWE is in `./dapps/web3modal/next`. In order to test this project, you need to set 2 environment variables 

`NEXT_PUBLIC_PROJECT_ID` and `NEXTAUTH_SECRET`

1. Navigate to demo project

```sh
cd /dapps/web3modal/next
# set .env variables
bun I
bun dev
```

2. Open your browser to `http://lcoalhost:3000`
3. Click "open modal"
4. Try and sign in with Ethereum 

<img width="512" alt="image" src="https://github.com/WalletConnect/web3modal/assets/1218847/57b7d49c-9ece-4c0a-968c-c74cd1ed3e60">

### List of related npm package versions

    "@tanstack/react-query": "5.18.1",
    "@web3modal/siwe": "^4.1.0",
    "@web3modal/wagmi": "4.0.0",
    "ethers": "^6.11.1",
    "next": "14.0.4",
    "next-auth": "^4.24.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "siwe": "^2.1.4",
    "viem": "2.7.6",
    "wagmi": "2.5.6"

**Comment by glitch-txs** [comment]
Will check, but works for me on labs: https://lab.web3modal.com/library/wagmi-siwe/

**Comment by glitch-txs** [comment]
Found the issue, you need to have `@web3modal/wagmi` and `@web3modal/siwe` in the same version. The problem is that they are pointing to the core library and the package manager installs the core twice, one for 4.0.0 and other one for 4.1.0, so you have a duplicate in packages, the store is in the core package so the address is undefined in one of them, which makes it fail.
```sh
bun add @web3modal/wagmi@latest @web3modal/siwe@latest
```
should fix it

**Comment by glitch-txs** [comment]
hmm found another issue with sub deps, will address

**Comment by joeblau** [comment]
I updated the libraries and it's still failing.

Where is the source code for the wage-siwe code? My guess is that this repo is not using NextAuth as documented in the wallet connect documentation.

https://docs.walletconnect.com/web3modal/nextjs/siwe/next-auth

If you run the sample code I've provided, you'll see the failure. I've also updated the dependencies on the sample project.

```
    "@tanstack/react-query": "5.18.1",
    "@web3modal/siwe": "latest",
    "@web3modal/wagmi": "latest",
    "ethers": "^6.11.1",
    "next": "14.0.4",
    "next-auth": "^4.24.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "siwe": "^2.1.4",
    "viem": "2.7.6",
    "wagmi": "2.5.6"
```

**Comment by joeblau** [comment]
Ah I figured out the final issue. The way I was using the `NextAuth` component, only captured the `NextAuthOptions` and not the request or response which means I couldn't get the csrf token from the headers.

I've made this change, and now the demo is working, but I'm going to re-factor my code to pass the request and response into the NextAuth component.  

```diff
- const nonce = await getCsrfToken({ req: { headers: req.headers } });
+ const nonce = cookies().get("next-auth.csrf-token")?.value.split("|")[0];
```

Thanks!

**Comment by ymonye** [comment]
Hi, your docs at:
https://docs.walletconnect.com/web3modal/nextjs/siwe/next-auth

They still reference:
`import type { SIWECreateMessageArgs, SIWESession, SIWEVerifyMessageArgs } from '@web3modal/core'`

Whereas only `@web3modal/siwe` has those exports:
`import { SIWECreateMessageArgs, SIWESession, SIWEVerifyMessageArgs } from '@web3modal/siwe'`

Also the entire `SIWE` guide is not as clear & straight-forward as the `Getting Started` guide. It would be helpful for ex explaining where to save configs from the `Configure your SIWE Client` section as it was done in the Getting Started guide, which made sense for me yet would be much smoother for more readers.

**Comment by wslyvh** [comment]
Hi @glitch-txs 
> Fix will be available in 4.1.1

I'm trying to upgrade from ^4.0.0, but this still seems to be an issue with @ 4.1.5
`yarn add @web3modal/siwe@latest` helps, but that isn't documented and seems unnecessary?

**Comment by glitch-txs** [comment]
@wslyvh hi, this is a issue with yarn specifically but it will be fixed in the next patch
